SwiftI18n/Classes/Main/I18nManager.swift

```diff
@@ -12,13 +12,27 @@ public class I18nManager {
 
     public static var instance = I18nManager()
     
+    public var overrideToFallbackLanguage = false
+    public var fallbackLanguage: String?
+    
     private init() {}
     
     public var defaultLanguage: String?
     public var availableLanguages: [String]?
     
-    public var localizationPerformingBlock: (_ key: String, _ language: String)->(String) = { (key, language) in
-        return NSLocalizedString(key, tableName: language, comment: "")
+    public lazy var localizationPerformingBlock: (_ key: String, _ language: String)->(String) = { [weak self] (key, language) in
```

Comment:
Hi @Antonijo2307 I don't think this is a good approach since it introduces `major chages` to the library.
`localizationPerformingBlock` is public and was designed to allow anyone to add set their own implementation.
--
In general what you are doing here can already be done in your project, no need to change this library.
--
If you want to, you can add it as a convenience  function that will set `localizationPerformingBlock` to what you need.

Something like:
```
public static func localizationPerformingBlockWithFallbackLanguage(fallbackLanguage: String) = { [weak self] (key, language) in
...
}
```
and then in your code you can simply do:
`Manager.localizationPerformingBlock = Manager.localizationPerformingBlockWithFallbackLanguage('your_language')`

Reply:
It was task requirement to add it to I18nManager.instance, I thought you all agreed on it. If this is not necessary just let me know and I will close this PR...

Reply:
Hm okay, let me go through this comment:
> In general what you are doing here can already be done in your project
The main problem is that no one is aware of this, and that's why we got the request from some projects. The issue is that the client doesn't want to see the translation key in places where the translation is missing for the production targets.
I'm okay with your proposal to create a new block, `localizationPerformingBlockWithFallbackLanguage,` which will generate practically the same outcome. Replace missing translations with some language (usually the default one).

> to not pollute the library with project specific things
It was mentioned in a few projects, so it's more than one specific project. We want to push our teams to start doing this: showing the key on the screen on staging builds and showing the default language translation on production.

> If you need to, you can add it as a convenience function that will set localizationPerformingBlock to what you need.
> ...
> and then in your code you can simply do
If the output is the same, and it's easy for the lib user as you defined it here, then I'm okay with that approach! Let's do it that way, then, without the proposed properties.

Reply:
> We want to push our teams to start doing this: showing the key on the screen on staging builds and showing the default language translation on production.

Hey @jabou , I don't think this is a good idea. It seems you are trying to solve a wrong problem.
Having a fallback translation instead of the correct one is still a bug.

The problem you have is that not all the translation are filled in.
The way to solve it is to make sure that all the translations are filled in :) 

This is something that should be addressed, for example: "Check all translations are added" button can be added to polyglot web, and your project's release process should include someone pressing that button.

Reply:
I totally agree with you that this should not end in production, but that's not how some of our clients work. 

To mention a few use cases we are trying to cover with this:
1. Translation is added for a future release and is not currently used in the app. The button will not cover that case, as it's okay not to have the translation at this moment.
2. The client has an external company that adds missing translations. Sometimes, that company deletes a translation or forgets to add it. In that case, rather than showing `some_translation_key` on screen, most apps show the "default" (usually English) translation as a mix of two languages (still not great, but much better than showing just a raw translation key). I once even saw a mix of German and English words in Apple settings (places where German translation was missing). That's something we want to accomplish; if someone breaks something with translation, we want to fall back to the default language on production, while on staging builds, we want to show the translation key for easier debugging. Of course, with the flexibility to always show keys if the client wants or to always show fallback if that's the case they want on staging.

_We will not cover the case if someone deliberately puts the translation key value under the default translation box._

Reply:
@Antonijo2307, can you please check those comments and give us your input? Regarding the braking change, I agree with Vlaho that we should not change public methods/properties in this way, as this could potentially break someone's active implementation. 

Please check if you can change how this can be handled from the comments Vlaho left in a thread (e.g. `localizationPerformingBlockWithFallbackLanguage` approach).

Reply:
Ok, we decided not to touch the block it self but to implement the changes in method that calls that block.